misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/UrlFinder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina;
+
+import io.ballerina.compiler.syntax.tree.BasicLiteralNode;
+import io.ballerina.compiler.syntax.tree.NamedArgumentNode;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.ParenthesizedArgList;
+import io.ballerina.compiler.syntax.tree.PositionalArgumentNode;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.Token;
+
+/**
+ * Visitor to file the base url of endpoint.
+ *
+ * @since 2.0.0
+ */
+public class UrlFinder extends NodeVisitor {
+
+    private String token;
+
+    public String getToken() {
+
+        return token;
+    }
+
+    @Override
+    public void visit(NamedArgumentNode namedArgumentNode) {
+
+        visitSyntaxNode(namedArgumentNode);
+
+    }
+
+    @Override
+    public void visit(PositionalArgumentNode positionalArgumentNode) {
+
+        visitSyntaxNode(positionalArgumentNode);
+
+    }
+
+    @Override
+    public void visit(SimpleNameReferenceNode simpleNameReferenceNode) {
+
+        visitSyntaxNode(simpleNameReferenceNode);
+
+    }
+
+    @Override
+    public void visit(QualifiedNameReferenceNode qualifiedNameReferenceNode) {
+
+        visitSyntaxNode(qualifiedNameReferenceNode);
+
+    }
+
+    @Override
+    public void visit(BasicLiteralNode basicLiteralNode) {
+
+        visitSyntaxNode(basicLiteralNode);
+    }
+
+    @Override
+    public void visit(ParenthesizedArgList parenthesizedArgList) {
+
+        visitSyntaxNode(parenthesizedArgList);
+
+    }
+
+    @Override
+    public void visit(Token token) {
+
+        this.token = token.text();
```

Comment:
Here we assume the `new()` expression has only a single parameter and it's the URL. Can that be assumed?